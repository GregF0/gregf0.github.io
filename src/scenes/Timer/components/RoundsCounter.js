import React from 'react'
import styled from 'styled-components'
import { useSelector } from 'react-redux'

export const RoundsCounter = () => {
  const { currentRound, timeLeft } = useSelector((state) => state.timer)
  const { rounds } = useSelector((state) => state.settings)

  return (
    <CounterLabel>
      {timeLeft && currentRound}/{timeLeft && rounds}
    </CounterLabel>
  )
}

export const CounterLabel = styled.span`
  font-weight: bold;
`
