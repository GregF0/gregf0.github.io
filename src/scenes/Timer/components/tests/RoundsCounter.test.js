import React from 'react'
import * as redux from 'react-redux'
import { createShallow } from '@material-ui/core/test-utils'
import { RoundsCounter, CounterLabel } from '../RoundsCounter'

describe('<RoundsCounter />', () => {
  const shallow = createShallow()
  const createWrapper = () => {
    return shallow(<RoundsCounter />)
  }

  const createStore = () => {
    const store = {
      timer: {
        currentRound: 3,
        timeLeft: { minutes: 25, seconds: 0 },
      },
      settings: { rounds: 4 },
    }

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((callback) => callback(store))

    return store
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should render <RoundsCounter /> correctly', () => {
    createStore()
    expect(createWrapper()).toMatchSnapshot()
  })

  test('should render current rounds/total rounds', () => {
    createStore()
    expect(createWrapper().find(CounterLabel).text()).toBe('3/4')
  })
})
