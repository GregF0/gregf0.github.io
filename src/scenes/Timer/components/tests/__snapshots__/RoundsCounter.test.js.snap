// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundsCounter /> should render <RoundsCounter /> correctly 1`] = `
<styled.span>
  3
  /
  4
</styled.span>
`;
